### DISCUSSION TITLE
The metrics reported at the same time are stored in one table ？

### DISCUSSION THREAD

**Discussion by cxntsh** [apache/skywalking#10090] [Ideas]
At present, the storage model of Skywalking OAP service is an indicator and a table. These indicators may come from Service, ServiceInstance, ServiceEndpoint, etc. Is there such a possibility? L2 will perform a one-time merge in the future, and only generate one insert SQL and store it in a wide table, because the id, service_id, entity_id, and time_buket of these data are all the same. Why do we have to think like this, because we found that these fields in the entire database occupy a lot of space, and the actual data storage only occupies a small part, such as value, percentage, numerator and other fields. After all, the MetricsPersistentWorker class is an instance of a metric, so I think it has the basis for this implementation. Teacher Wu Sheng, what do you think?  @wu-sheng

**Comment by cxntsh** [comment]
Combined with my above ideas, if we use a columnar storage database, such as Clickhouse, then the storage space of this data can be greatly reduced

**Comment by wu-sheng** [comment]
I don't know why you ping me, there's nothing relative for me.

**Reply by cxntsh** [reply]
I just want to ask Teacher Wu Sheng to help verify whether my idea is feasible or whether I think it is a good idea ？

**Reply by wu-sheng** [reply]
There is no maintainer working on clickhouse. So, I don't know how to response.
If you have deployment architecture with solid benchmarks to support your proposal, we could start the discussion.
